### ISSUE TITLE
Retry options seem to be deprecated, but not according to the docs

### ISSUE THREAD

**Issue by ingnil** [google-github-actions/auth#391]
### TL;DR

Using retry options give warnings about the options being deprecated, but there is no mention of this in the README.

### Expected behavior

Either the options should not give warnings, or the deprecation should be mentioned in the README. Ideally, the docs should be accompanied by a suggested alternate solution, because retry functionality is useful. Authentication using Workload Identity Federation sometimes fail spuriously* (which then succeeds when the workflow is rerun), so decreasing this probability with retries is a useful feature.

*= We did not see these spurious failures when authenticating with static JSON keys (which we did previously), but using static JSON keys is no longer the recommended authentication method.

### Action YAML

```yaml
- name: Authenticate to Google Cloud
  uses: google-github-actions/auth@v2
  with:
    workload_identity_provider: "${{ vars.WIF_PROVIDER }}"
    service_account: "${{ vars.WIF_SERVICE_ACCOUNT }}"
    retries: 5
    backoff: 1000
```

### Log output

```text
Warning: Input 'retries' has been deprecated with message: This field is no longer used and will be removed in a future release.
Warning: Input 'backoff' has been deprecated with message: This field is no longer used and will be removed in a future release.
```
